core/shared/src/main/scala/zio/Cause.scala

```diff
@@ -140,6 +140,36 @@ sealed trait Cause[+E] extends Product with Serializable { self =>
       case (_, Interrupt(_))    => false
     }
 
+  final def stripTopSilent: Cause[E] =
+    self match {
+      case Meta(cause, Data(_, true)) => cause
+      case other                      => other
+    }
+
+  final def stripSilent: Option[Cause[E]] =
+    self match {
+      case Meta(_, Data(_, true)) => None
```

Comment:
To be compositional we want to say that we won't report children of silent nodes unless a lower level `Meta` indicates that they should not be silent. So for example if we have:

```scala
Cause.silent(Cause.Both(Cause.Fail("fail1"), Cause.nonSilenct(Cause.Fail("fail2')))
```

`stripSilent` should return `Some(Cause.Fail("fail2"))`. So we need to recurse in the `Meta(_, true)` case.